maven/plugins/vrealize/src/main/java/com/vmware/pscoe/maven/plugins/PushMojo.java

```diff
@@ -54,30 +54,45 @@ public class PushMojo extends AbstractIacMojo {
 	private List<String> filesChanged;
 
 	private static Package packageFromArtifact(Artifact artifact) {
-		return PackageFactory.getInstance(PackageType.fromExtension(artifact.getType()), artifact.getFile(), new MavenArtifactPackageInfoProvider(artifact).getPackageName());
+		return PackageFactory.getInstance(PackageType.fromExtension(artifact.getType()), artifact.getFile(),
+				new MavenArtifactPackageInfoProvider(artifact).getPackageName());
 	}
 
-	private void importArtifacts(Collection<Artifact> allArtifacts) throws MojoExecutionException {
-		Map<String, List<Artifact>> artifactsByType = allArtifacts.stream().collect(Collectors.groupingBy(Artifact::getType));
-		
+	private void importArtifacts(Collection<Artifact> allArtifacts)
+			throws MojoExecutionException, ConfigurationException {
+		Map<String, List<Artifact>> artifactsByType = allArtifacts.stream()
+				.collect(Collectors.groupingBy(Artifact::getType));
+
 		for (Map.Entry<String, List<Artifact>> type : artifactsByType.entrySet()) {
 			PackageType pkgType = PackageType.fromExtension(type.getKey());
 			if (pkgType == null) {
 				continue;
 			}
 			try {
-				List<Package> packages = artifactsByType.get(type.getKey()).stream().map(PushMojo::packageFromArtifact).collect(Collectors.toList());
+				List<Package> packages = artifactsByType.get(type.getKey()).stream().map(PushMojo::packageFromArtifact)
+						.collect(Collectors.toList());
 				PackageStore<?> store = getConfigurationForType(PackageType.fromExtension(type.getKey()))
-						.flatMap(configuration -> Optional.of(PackageStoreFactory.getInstance(configuration))).orElseThrow(() -> new ConfigurationException(
-								"Unable to find PackageStore based on configuration. Make sure there is configuration for type: " + pkgType.name()));
+						.flatMap(configuration -> PackageStoreFactory.getInstanceWrapped(configuration))
```

Comment:
I get it, the whole idea behind getInstanceWrapped. TBF, I prefer it (using RuntimeException), :D Can't we do it everywhere?

Reply:
I am just desperate to clean up this "Error Propagation" or whatever you want to call it